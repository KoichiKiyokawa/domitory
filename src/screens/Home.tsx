import React from 'react'
import {
  Card,
  CardItem,
  Container,
  Content,
  Icon,
  Text,
  Right
} from 'native-base'
import { FontAwesome5 } from '@expo/vector-icons'

import { RootContext } from '../contexts/RootContext'
import { parseToMonthWithDay } from '../utils/dateUtil'

export default () => {
  const { notices, setNotices } = React.useContext(RootContext)

  React.useEffect(() => {
    setNotices([
      { text: '自転車が盗まれました', date: new Date() },
      { text: '近隣住民から苦情がきました', date: new Date() }
    ])
  }, [])

  return (
    <Container>
      <Content>
        {/* お知らせ */}
        <Card>
          <CardItem header>
            <FontAwesome5
              name="bullhorn"
              color="teal"
              size={24}
              style={{ marginRight: 10 }}
            />
            <Text>お知らせ</Text>
          </CardItem>
          {notices.map((notice, i) => (
            <CardItem key={i}>
              <Text style={{ marginRight: 10, color: 'silver' }}>
                {parseToMonthWithDay(notice.date)}
              </Text>
              <Text>{notice.text}</Text>
              <Right>
                <Icon name="arrow-forward" />
              </Right>
            </CardItem>
          ))}
        </Card>
      </Content>
    </Container>
  )
}
