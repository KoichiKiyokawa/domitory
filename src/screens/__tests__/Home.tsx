import React from 'react'
import Home from '../Home'

import renderer from 'react-test-renderer'

describe('easy test', () => {
  test('plus', () => {
    expect(1 + 1).toBe(2)
  })
})

describe('rendering test', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
