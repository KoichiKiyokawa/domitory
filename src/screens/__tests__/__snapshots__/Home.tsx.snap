// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          header={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": null,
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            管理人さん情報
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              管理人は現在...
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flex": 1,
                  },
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f4f4f4",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 57,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": undefined,
                    "shadowOffset": undefined,
                    "shadowOpacity": undefined,
                    "shadowRadius": undefined,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16.5,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  管理室にいます
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#d9534f",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 57,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "marginLeft": 10,
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": undefined,
                    "shadowOffset": undefined,
                    "shadowOpacity": undefined,
                    "shadowRadius": undefined,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontFamily": "System",
                        "fontSize": 16.5,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  出かけています
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          header={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": null,
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
            uppercase={false}
          >
            お知らせ
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
